controllers/fenceagentsremediation_controller_test.go

```diff
@@ -582,15 +582,20 @@ func cleanupFar() func(ctx context.Context, far *v1alpha1.FenceAgentsRemediation
 			return err
 		}
 
+		var isCRNotFound bool
 		ConsistentlyWithOffset(1, func() error {
 			deleteErr := k8sClient.Get(ctx, client.ObjectKeyFromObject(far), cr)
 			if apierrors.IsNotFound(deleteErr) {
 				// when trying to create far CR with invalid name
 				log.Info("Cleanup: Got error 404", "name", cr.Name)
+				isCRNotFound = true
 				return nil
 			}
 			return deleteErr
 		}, pollInterval, timeoutPostRemediation).Should(BeNil(), "CR should be deleted")
+		if !isCRNotFound {
```

Comment:
I won't block the PR on this, but I found a bit tricky to read the test if some checks are "hidden" in clean up functions

Reply:
ok, it can be moved to `DeferCleanup`